File: 6-kyu/consecutive-strings/swift/solution.swift
func longestConsec(_ strarr: [String], _ k: Int) -> String {
    guard !strarr.isEmpty && k <= strarr.count && k > 0 else { return "" }
    var ret = [String]()
    for i in 0..<strarr.count {
        var str: String = ""
        if i + k < strarr.count {
            str = strarr[i..<(i + k)].joined()
        } else {
            str = strarr[i..<strarr.count].joined()
        }
        ret.append(str)
    }
    if let result = ret.max(by: {$0.characters.count < $1.characters.count }) {
        return result
    } else {
        return ""
    }
}

File: 6-kyu/consecutive-strings/python/solution.py
def longest_consec(strarr, k):
    if len(strarr) == 0 or k > len(strarr) or k <= 0:
        return ''
    longest = ''
    for i in range(len(strarr)):
        curr = ''
        if i + k < len(strarr):
            curr = ''.join(strarr[i:i + k])
        else:
            curr = ''.join(strarr[i:])
        if len(curr) > len(longest):
            longest = curr
    return longest
